### PR TITLE
Kubeadm - add all subcommands to phases

### LINKED ISSUE #1243: phase flags that don't make sense
Currently global init flags show up on some phase sub-commands where they don't apply or make any sense.  

in the example below: 
```
 --apiserver-advertise-address string   The IP address the API Server will advertise it's listening on. Specify '0.0.0.0' to use the address of the default network interface.
      --apiserver-bind-port int32            Port for the API Server to bind to. (default 6443) 
```

Don't make sense in the context of the phase. 

``` 
$ kubeadm init phase addon --help
Installs required addons for passing Conformance tests

Usage:
  kubeadm init phase addon [flags]
  kubeadm init phase addon [command]

Available Commands:
  coredns     Installs the CoreDNS addon to a Kubernetes cluster
  kube-proxy  Installs the kube-proxy addon to a Kubernetes cluster

Flags:
      --apiserver-advertise-address string   The IP address the API Server will advertise it's listening on. Specify '0.0.0.0' to use the address of the default network interface.
      --apiserver-bind-port int32            Port for the API Server to bind to. (default 6443)
      --config string                        Path to kubeadm config file. WARNING: Usage of a configuration file is experimental.
      --feature-gates string                 A set of key=value pairs that describe feature gates for various features. Options are:

  -h, --help                                 help for addon
      --image-repository string              Choose a container registry to pull control plane images from (default "k8s.gcr.io")
      --kubeconfig string                    The kubeconfig file to use when talking to the cluster. If the flag is not set, a set of standard locations are searched for an existing KubeConfig file. (default "/etc/kubernetes/admin.conf")
      --kubernetes-version string            Choose a specific Kubernetes version for the control plane. (default "stable-1")
      --pod-network-cidr string              Specify range of IP addresses for the pod network. If set, the control plane will automatically allocate CIDRs for every node.
      --service-cidr string                  Use alternative range of IP address for service VIPs. (default "10.96.0.0/12")
      --service-dns-domain string            Use alternative domain for services, e.g. "myorg.internal". (default "cluster.local")

Global Flags:
      --log-file string   If non-empty, use this log file
      --rootfs string     [EXPERIMENTAL] The path to the 'real' host root filesystem.
      --skip-headers      If true, avoid header prefixes in the log messages
  -v, --v Level           log level for V logs
``` 

/assign @fabriziopandini @neolit123  @timothysc 
/cc @rdodev

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
This PR makes phase subcommands that are not "leaves of the tree" print "This command is not meant to be run on its own. See list of available subcommands." + helps.

If the developer wants to provide to the user the "all" behavior, it can be done by implementing a phase with a special flag. Such phase will not be considered in the Runner workflow, but will generate a subcommand as expected.

the developer remains in full control of flags assigned to all phases.

**Which issue(s) this PR fixes** : 
Fixes: https://github.com/kubernetes/kubeadm/issues/1243

**Special notes for your reviewer**:
Depends on:
- [x] https://github.com/kubernetes/kubernetes/pull/71066

/sig cluster-lifecycle
/kind bug
/priority critical-urgent
/milestone v1.13
/assign @timothysc
/assign @neolit123 
@kubernetes/sig-cluster-lifecycle-pr-reviews 
@liztio 

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: cmd/kubeadm/app/cmd/phases/uploadconfig.go [hunk 1]
```diff
@@ -71,6 +71,11 @@ func NewUploadConfigPhase() workflow.Phase {
 		Short:   "Uploads the kubeadm and kubelet configuration to a ConfigMap",
 		Long:    cmdutil.MacroCommandLongDescription,
 		Phases: []workflow.Phase{
+			{
+				Name:           "all",
+				Short:          "Uploads all configuration to a config map",
+				RunAllSiblings: true,
```
Comment (yagonobre) [inline cmd/kubeadm/app/cmd/phases/uploadconfig.go]: Add `InheritFlags: getUploadConfigPhaseFlags(),`

**Review by yagonobre — APPROVED** [review]
Good work!

**Review by neolit123 — APPROVED** [review]
thanks @fabriziopandini 

/lgtm
/hold

File: cmd/kubeadm/app/cmd/phases/uploadconfig.go (hunk 1 above)
Comment (neolit123) [reply cmd/kubeadm/app/cmd/phases/uploadconfig.go]: actually yes, this this seems correct.
thanks @yagonobre

File: cmd/kubeadm/app/cmd/phases/workflow/runner.go [hunk 2]
```diff
@@ -211,6 +211,12 @@ func (e *Runner) Run() error {
 			return nil
 		}
 
+		// Errors if phases that are meant to create special subcommands only
+		// are wrongly assigned Run Methods
+		if p.RunAllSiblings && (p.RunIf != nil || p.Run != nil) {
```
Comment (timothysc) [inline cmd/kubeadm/app/cmd/phases/workflow/runner.go]: We need a test for the phase runner changes, otherwise LGTM

**Review by timothysc — COMMENTED** [review]
/approve

**Comment by AishSundar** [comment]
@fabianofranz @neolit123 @timothysc how confident are we in getting this in by Code freeze tomorrow?

**Comment by neolit123** [comment]
@AishSundar pretty confident, no worries.
also this is a critical-bug fix at this point and it can go in code freeze.

**Comment by fabriziopandini** [comment]
/hold
Waiting for last feedback from @liztio

**Review by liztio — APPROVED** [review]
I don't love the UX (the `all` command seems less user-friendly, and the RunSiblingCommands magic flag feels super awkward to me), but I recognise this as the most expedient solution, and an incremental step towards better UX.

**Comment by neolit123** [comment]
@liztio if we don't add all flags to either the parent phase or to a "all" phase we don't have a way to expose to the user a way to control only certain features:
```
kubeadm phase certs --some-custom-flag (this is not good, due to an issue with alternating flags and sub-commands
or
kubeadm phase certs all --some-custom-flag
```
so if we remove `--some-custom-flag` from `all` or `certs` the user has to break the whole phase into subphases if they want to apply `--some-custom-flag` to a couple of sub-phases:
```
kubeadm phase certs sub-phase1
kubeadm phase certs sub-phase2 --some-custom-flag
kubeadm phase certs sub-phase3
kubeadm phase certs sub-phase4 --some-custom-flag
kubeadm phase certs sub-phase5
...
```
a hope that explains the motivation here?

**Comment by nikopen** [comment]
@pwittrock @thockin @smarterclayton for approval (docs)

**Review by timothysc — APPROVED** [review]
/lgtm

**Pull request merged**